spec/services/claims/claim/remove_mentor_training_hours_spec.rb

```diff
@@ -0,0 +1,32 @@
+require "rails_helper"
+
+describe Claims::Claim::RemoveMentorTrainingHours do
+  subject(:service) { described_class.call(claim:) }
+
+  let!(:claim) { create(:claim, reference: nil, status: :internal, school:) }
+  let(:school) { create(:claims_school, urn: "1234") }
+
+  it_behaves_like "a service object" do
+    let(:params) { { claim: } }
+  end
+
+  describe "#call" do
+    it "removes all mentor trainings without hours from a claim" do
+      create(:mentor_training, hours_completed: nil, claim:)
+      create(:mentor_training, hours_completed: nil, claim:)
+      expect { service }.to change(Claims::MentorTraining, :count).by(-2)
+
+      expect(claim.mentor_trainings).to eq([])
+    end
+
+    context "when mentor trainings have hours completed" do
+      it "does not remove mentor trainings with hours completed from a claim" do
+        training1 = create(:mentor_training, hours_completed: 10, claim:)
+        training2 = create(:mentor_training, hours_completed: 10, claim:)
+        expect { service }.to change(Claims::MentorTraining, :count).by(0)
+
+        expect(claim.mentor_trainings).to eq([training1, training2])
+      end
+    end
+  end
```

Comment:
Can we add a case where it removes empty mentor trainings but keeps those with hours belonging to a single claim?

These test cases give me confidence that they remove empty-hour records entirely, or not at all, but it doesn't give me confidence in the mixed records case.